crypto/s2n_certificate.c

```diff
@@ -823,7 +823,7 @@ static int s2n_parse_x509_extension(struct s2n_cert *cert, const uint8_t *oid,
             /* ASN1_STRING_length() returns the length of the content of `asn1_str`.
             * Ref: https://www.openssl.org/docs/man1.1.0/man3/ASN1_STRING_length.html.
             */
-            int len = ASN1_STRING_length(asn1_str);
+            uint32_t len = ASN1_STRING_length(asn1_str);
```

Comment:
This returns a `int` so it's best to keep it signed, or at least make sure it's non-negative before casting it.